validator/src/consensus/client.ts

```diff
@@ -1,126 +1,237 @@
-import type { Address } from "viem";
-import type {
-	FrostPoint,
-	GroupId,
-	ProofOfAttestationParticipation,
-	ProofOfKnowledge,
-} from "../frost/types.js";
+import type { Address, Hex } from "viem";
+import type { FrostPoint, GroupId, ProofOfKnowledge } from "../frost/types.js";
+// TODO: replace by final math logic imports
+import { FROST_MATH } from "../frost/types.js";
 import {
 	calculateParticipantsRoot,
 	generateParticipantProof,
 } from "./merkle.js";
-import type { Participant } from "./types.js";
+import type { FrostCoordinator, Participant } from "./types.js";
 
-type KeyGenInfo = {
-	participantsRoot: string;
+type KeygenInfo = {
+	groupId: GroupId;
 	participants: Participant[];
+	coefficients: bigint[];
+	participantIndex: bigint;
+	commitments: Map<bigint, FrostPoint[]>;
+	secretShares: Map<bigint, bigint>;
+	verificationShare?: FrostPoint;
+	signingShare?: bigint;
 };
 
-type KeyGenMaterial = {
-	coefficients: bigint[];
-	commitments: FrostPoint[];
+const findParticipationIndex = (
+	participants: Participant[],
+	validatorAddress: Address,
+): [bigint, number] => {
+	const arrayIndex = participants.findIndex(
+		(p) => p.address === validatorAddress,
+	);
+	if (arrayIndex < 0)
+		throw Error("Cannot find participant in participants array!");
+	const participantIndex = participants.at(arrayIndex)?.index;
+	if (participantIndex === undefined)
+		throw Error("Cannot determine FROST index of participant!");
+	return [participantIndex, arrayIndex];
```

Comment:
Not a comment on your code specifically, but I find "index" to be an overloaded term which causes confusion (does it start at 0? Is it a participant index that starts at 1? etc.)

I would propose that we change this to `Identifier` (this is what the [reference FROST implementation calls it](https://docs.rs/frost-core/latest/frost_core/struct.Identifier.html), which would give us clearer names for participants "start from 1" indexes/identifiers, from array "start from 0" indexes. I would propose renaming this in the contract as well.

Obviously, not for this PR, just something to help with code clarity going forward.